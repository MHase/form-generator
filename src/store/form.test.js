import reducer, { updateField, clearValues, initialState } from './form';

describe('Form reducer', () => {
  const state = {
    values: {
      name: '',
    },
  };

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });

  it(`handles UPDATE_FORM correctly`, () => {
    const payload = { name: 'Maciek', nickname: 'Coder' };
    const actual = reducer(state, updateField(payload));

    expect(actual).toEqual({
      values: {
        name: 'Maciek',
        nickname: 'Coder'
      }
    });
  });

  it(`handles RESET_FORM correctly`, () => {
    const actual = reducer(state, clearValues());

    expect(actual).toEqual(initialState);
  });
});
