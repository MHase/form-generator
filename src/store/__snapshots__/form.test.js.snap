// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form reducer handles RESET_FORM correctly 1`] = `
Object {
  "structure": Object {
    "Email": Object {
      "name": "Email",
      "required": true,
      "type": "email",
    },
    "Field": Object {
      "name": "Field",
      "options": Array [
        Object {
          "label": "IT",
          "value": "it",
        },
        Object {
          "label": "Product",
          "value": "product",
        },
        Object {
          "label": "Content",
          "value": "content",
        },
      ],
      "required": true,
      "type": "select",
    },
    "Name": Object {
      "minLength": 4,
      "name": "Name",
      "required": true,
      "type": "text",
    },
    "Nickname": Object {
      "name": "Nickname",
      "type": "text",
    },
    "Position": Object {
      "dependant": "Field",
      "name": "Position",
      "options": Object {
        "content": Array [
          Object {
            "label": "Junior Copywriter",
            "value": "jr_copy",
          },
          Object {
            "label": "Senior Copywriter",
            "value": "sr_copy",
          },
        ],
        "it": Array [
          Object {
            "label": "Front-end developer",
            "value": "fe_dev",
          },
          Object {
            "label": "Back-end developer",
            "value": "be_dev",
          },
          Object {
            "label": "Devops",
            "value": "devops",
          },
          Object {
            "label": "Webmaster",
            "value": "webmaster",
          },
        ],
        "product": Array [
          Object {
            "label": "Product owner",
            "value": "prod_owner",
          },
          Object {
            "label": "UX Designer",
            "value": "ux_des",
          },
          Object {
            "label": "UI Designer",
            "value": "ui_des",
          },
        ],
      },
      "required": true,
      "type": "select",
    },
  },
  "values": Object {
    "Email": "",
    "Field": "",
    "Name": "",
    "Nickname": "",
    "Position": "",
  },
}
`;

exports[`Form reducer handles UPDATE_FORM correctly 1`] = `
Object {
  "structure": Object {
    "Email": Object {
      "name": "Email",
      "required": true,
      "type": "email",
    },
    "Field": Object {
      "name": "Field",
      "options": Array [
        Object {
          "label": "IT",
          "value": "it",
        },
        Object {
          "label": "Product",
          "value": "product",
        },
        Object {
          "label": "Content",
          "value": "content",
        },
      ],
      "required": true,
      "type": "select",
    },
    "Name": Object {
      "minLength": 4,
      "name": "Name",
      "required": true,
      "type": "text",
    },
    "Nickname": Object {
      "name": "Nickname",
      "type": "text",
    },
    "Position": Object {
      "dependant": "Field",
      "name": "Position",
      "options": Object {
        "content": Array [
          Object {
            "label": "Junior Copywriter",
            "value": "jr_copy",
          },
          Object {
            "label": "Senior Copywriter",
            "value": "sr_copy",
          },
        ],
        "it": Array [
          Object {
            "label": "Front-end developer",
            "value": "fe_dev",
          },
          Object {
            "label": "Back-end developer",
            "value": "be_dev",
          },
          Object {
            "label": "Devops",
            "value": "devops",
          },
          Object {
            "label": "Webmaster",
            "value": "webmaster",
          },
        ],
        "product": Array [
          Object {
            "label": "Product owner",
            "value": "prod_owner",
          },
          Object {
            "label": "UX Designer",
            "value": "ux_des",
          },
          Object {
            "label": "UI Designer",
            "value": "ui_des",
          },
        ],
      },
      "required": true,
      "type": "select",
    },
  },
  "values": Object {
    "Email": "",
    "Field": "",
    "Name": "",
    "Nickname": "",
    "Position": "",
    "name": "Maciek",
  },
}
`;

exports[`Form reducer should return the initial state 1`] = `
Object {
  "structure": Object {
    "Email": Object {
      "name": "Email",
      "required": true,
      "type": "email",
    },
    "Field": Object {
      "name": "Field",
      "options": Array [
        Object {
          "label": "IT",
          "value": "it",
        },
        Object {
          "label": "Product",
          "value": "product",
        },
        Object {
          "label": "Content",
          "value": "content",
        },
      ],
      "required": true,
      "type": "select",
    },
    "Name": Object {
      "minLength": 4,
      "name": "Name",
      "required": true,
      "type": "text",
    },
    "Nickname": Object {
      "name": "Nickname",
      "type": "text",
    },
    "Position": Object {
      "dependant": "Field",
      "name": "Position",
      "options": Object {
        "content": Array [
          Object {
            "label": "Junior Copywriter",
            "value": "jr_copy",
          },
          Object {
            "label": "Senior Copywriter",
            "value": "sr_copy",
          },
        ],
        "it": Array [
          Object {
            "label": "Front-end developer",
            "value": "fe_dev",
          },
          Object {
            "label": "Back-end developer",
            "value": "be_dev",
          },
          Object {
            "label": "Devops",
            "value": "devops",
          },
          Object {
            "label": "Webmaster",
            "value": "webmaster",
          },
        ],
        "product": Array [
          Object {
            "label": "Product owner",
            "value": "prod_owner",
          },
          Object {
            "label": "UX Designer",
            "value": "ux_des",
          },
          Object {
            "label": "UI Designer",
            "value": "ui_des",
          },
        ],
      },
      "required": true,
      "type": "select",
    },
  },
  "values": Object {
    "Email": "",
    "Field": "",
    "Name": "",
    "Nickname": "",
    "Position": "",
  },
}
`;
